src/exampleWorkerManager.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-env browser */
+
+'use strict';
+
+// TODO: handle invalid grammar in textbox
+// TODO: fix needed eamples bug (need force update to get rid of ident)
+
+(function(root, initModule) {
+  if (typeof exports === 'object') {
+    module.exports = initModule;
+  } else {
+    root.exampleWorkerManager = initModule(root.ohm, root.ohmEditor,
+                                           root.utils, root.CheckedEmitter);
+  }
+})(this, function(ohm, ohmEditor, utils, CheckedEmitter) {
+  var exampleWorkerManager = new CheckedEmitter();
+
+  exampleWorkerManager.registerEvents({
+    'received:examples': ['ruleName', 'examples'],
+    'received:neededExamples': ['neededExamples']
+  });
+
+  var eventsToEmit = ['received:examples', 'received:neededExamples'];
+
+  var exampleWorker = new Worker('exampleWorker.js');
+  /* eslint-disable no-unused-vars */
+  var examplesNeeded = null;
+  /* eslint-enable no-unused-vars */
+
+  // TODO: may want to reset current worker instead
+
+  var grammar;
+
+  function resetWorker(grammar) {
+    if (exampleWorker) {
+      exampleWorker.terminate();
+    }
+    exampleWorker = new Worker('exampleWorker.js');
+    exampleWorker.onmessage = onWorkerMessage;
+    exampleWorker.postMessage({
+      name: 'initialize', recipe: grammar.toRecipe()
+    });
+
+    var examples = ohmEditor.examples.getExamples();
+    Object.keys(examples).forEach(function(id) {
+      var example = examples[id];
+      if (example.startRule) {
+        var match = grammar.match(example.text, example.startRule);
+      } else {
+        match = grammar.match(example.text);
+      }
+
+      if (match.succeeded()) {
+        exampleWorkerManager.addUserExample(example.startRule || grammar.defaultStartRule,
+                                            example.text);
+      }
+    });
+  }
+
+  ohmEditor.addListener('parse:grammar', function(_, g, __) {
+    grammar = g;
```

Comment:
Do you need to store this? It should be always available in `ohmEditor.grammar`, no?